### PR TITLE
common-kernel-build-errors.rst: Add systemd 233+ fix for 3.4 kernels

### PR DESCRIPTION
**Pull request by Herrie82**
Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

### CODE REVIEW

**Comment by bhush9** [comment]
This shouldn't be in common-errors, but more like part of kernel adjustments.. probably during initial kernel config adjustments?

File: porting/common-kernel-build-errors.rst [hunk 1]
```diff
@@ -77,10 +77,10 @@ Somehow, the implementation of the ``/proc`` filesystem is incomplete in some 3.
    /../../../../kernel/fairphone/msm8974/kernel/pid.c:81:15: error: 'PROC_PID_INIT_INO' undeclared here (not in a function)
      .proc_inum = PROC_PID_INIT_INO,
 
-Add the following line after all of the other ``#include`` lines in the file::
+Add the following line after all of the other ``#include`` lines in the file kernel/user_namespace.c similar to https://github.com/Halium/android_kernel_lge_hammerhead/commit/5754614eb43dea44a99e54898e3b83d4d96d8b83 ::
 
    #include <linux/proc_fs.h>
-
+
```
Comment (bhush9) [inline porting/common-kernel-build-errors.rst]: huh?

Comment (Herrie82) [reply porting/common-kernel-build-errors.rst]: @bhush9 Ah missed that one, pushed a cleanup

**Comment by doniks** [comment]
does this patch affect the "3.10 statement" here? https://docs.halium.org/en/latest/porting/first-steps.html#pick-an-android-target-device

**Comment by Herrie82** [comment]
@doniks We haven't tested Halium as such at our end, however we have Mako, Hammerhead, Tenderloin and Onyx (all 3.4 kernel) working on LuneOS which uses (a patched) systemd 237 (from Yocto https://github.com/openembedded/openembedded-core/tree/sumo/meta/recipes-core/systemd/systemd) currently. It requires a few patches to the kernel for namespaces and fs mainly. It could be Halium rootfs will use more systemd features that require a higher kernel version. But for LuneOS with systemd, the patches to systemd from Yocto side and below kernel patches do the trick:

https://github.com/Halium/android_kernel_lge_hammerhead/commit/17a8d0630b424d66981900e701ae2a452d4b8726
https://github.com/Halium/android_kernel_lge_hammerhead/commit/5754614eb43dea44a99e54898e3b83d4d96d8b83
https://github.com/Halium/android_kernel_lge_hammerhead/commit/bd221854de33b75db7a7fa01cb34274b62a7cbf8
https://github.com/Halium/android_kernel_lge_hammerhead/commit/25437b2a54dd619a96e268ecaf303b089aa785e4

**Comment by Herrie82** [comment]
@doniks We ran into an issue with our Tenderloin target which used a newer 3.4 kernel release that used f[get|put]_light() in vfs_fstat() instead of fget_raw and fput and would cause the target not to boot. By reversing this commit: https://github.com/shr-distribution/linux/commit/01141a353241d545a2aaf9addaa32afe39a98562 we got it working again on Yocto Warrior with pretty much bleeding edge systemd 241 (including a bunch of patches provided by Yocto as usual): https://github.com/openembedded/openembedded-core/tree/warrior/meta/recipes-core/systemd

**Pull request merged**